src/Adapter/Tornado/Internal/PendingPromise.php

```diff
@@ -15,6 +15,19 @@ class PendingPromise implements Promise
     private $callbacks = [];
     private $isSettled = false;
     private $hasBeenYielded = false;
+    private $exception;
+
+    public function __destruct()
+    {
+        if (!$this->hasBeenYielded && $this->exception) {
+            throw $this->exception;
+        }
+    }
+
+    public function throwOnDestructIfNotYielded(\Throwable $throwable)
```

Comment:
In other adapters, you need a `$throwOnDestructIfNotYielded` boolean property, but not here? Not sure if it's equivalent…